crates/modelardb_compression/src/models/macaque_v.rs

```diff
@@ -150,6 +161,34 @@ impl Gorilla {
         self.update_min_max_and_last_value(value);
     }
 
+    /// Rewrite least mantissa bits of the `value` to zero within the `error_bound`.
+    fn rewrite_value_with_log_method(&self, value: Value) -> Value {
+        if value.abs() == 0.0 || value.is_infinite() || value.is_nan() {
+            return value;
+        }
+
+        let value_as_u32 = value.to_bits();
+        let abs_error_bound =
+            models::maximum_allowed_deviation(self.error_bound, value as f64) as f32;
+        let exponent = get_exponent(value);
+        let factorized_epsilon = abs_error_bound / 2f32.powi(exponent);
+        // Rewriting value using 23 - ⌈log2 factorized_epsilon⌉ bits never exceeds the error bound.
+        // However, one more bit can be rewritten when the majority of the least mantissa bits
+        // are 0. Thus we use 23 - ⌊log2 factorized_epsilon⌋ and then perform an extra check
+        // to ensure if the error bound is not exceeded.
+        let mut rewrite_position = 23 - factorized_epsilon.log2().abs().floor() as i32;
+        let mut rewritten_value = f32::from_bits(rewrite_bits_by_n(value_as_u32, rewrite_position));
+
+        // If the error bound is exceeded, value is rewritten with one less bit i.e.,
+        // using 23 − ⌈log2 factorized_epsilon⌉.
```

Comment:
But `23...` is the number of bits that exceeded the error bound, despite above stating that it cannot, so should it be `23 - ....` in all comments in this method or should the number of bits be different in some comments? If the number of should be the same in all comments, please expand the comments to describe why.

Reply:
The comment was extended to be more clear.